src/components/input.tsx

```diff
@@ -0,0 +1,64 @@
+import { type VariantProps, cva } from 'class-variance-authority'
+import React, { forwardRef } from 'react'
+
+import { cn } from '@/utils/cn'
+
+const inputVariants = cva(
+  [
+    'w-full h-14 rounded-xl border border-2 px-4 text-base text-medium',
+    'focus:outline-none focus:border-focus',
+    'placeholder-disabled',
+  ],
+
+  {
+    variants: {
+      variant: {
+        default: 'border-base',
+      },
+      error: {
+        true: 'border-error',
+      },
+      loading: {
+        true: 'text-transparent',
+      },
+      hasIcon: {
+        true: 'pl-12',
+      },
+      disabled: {
+        true: 'bg-disabled border-transparent',
+      },
+    },
+    defaultVariants: {
+      variant: 'default',
+    },
+  },
+)
+
+export interface InputProps
+  extends VariantProps<typeof inputVariants>,
+    React.InputHTMLAttributes<HTMLInputElement> {
+  loading?: boolean
+  error?: boolean
+  disabled?: boolean
+  icon?: React.ReactNode
+  type?: 'text' | 'password' | 'email' | 'number'
+}
+
+export const Input = forwardRef<HTMLInputElement, InputProps>(function Input(
+  { type = 'text', icon, loading, error, disabled, className, ...props },
+  ref,
+) {
+  return (
+    <div className="relative">
+      {icon && <div className="absolute left-4 top-4">{icon}</div>}
```

Comment:
Are we satisfied with the absolute positioning for the icon or should we inline it with the input?

Reply:
I think it's fine for now, I don't see yet an advantage of using it inline 🤔